images/ubuntu/editor/Dockerfile

```diff
@@ -171,6 +168,56 @@ RUN echo "$version-$module" | grep -q -vP '^20(?!18).*android' \
   # Update '~/.bashrc' to enable using variables when logging in
   && echo ". /usr/bin/unity-editor.d/android-2019+.sh" >> ~/.bashrc
 
+#=======================================================================================
+# [2021.x/2022.x-android] Set CMDLINE Tools Path
+#=======================================================================================
+RUN echo "$version-$module" | grep -q -vP '^(202[1-9]|20[3-9]).*android' \
+  && exit 0 \
+  || : \
+  && . ~/.bashrc \
+  && export RAW_CMDLINE_TOOLS_PATH=$(jq -cr '(.[] | select(.id | contains("android-sdk-command-line-tools"))).renameTo' $UNITY_PATH/modules.json) \
+  && export ESCAPED_UNITY_PATH=$(printf '%s' "$UNITY_PATH" | sed 's/[#\/]/\\\0/g') \
+  && export ANDROID_CMDLINE_TOOLS_PATH=$(echo $RAW_CMDLINE_TOOLS_PATH | sed -e "s/{UNITY_PATH}/$ESCAPED_UNITY_PATH/g") \
+  # Prefer cmdline tools over legacy tools
+  && export PATH=${ANDROID_CMDLINE_TOOLS_PATH}/bin:${PATH} \
+  && { \
+    echo "export ANDROID_SDK_ROOT=${ANDROID_SDK_ROOT}"; \
+    echo "export ANDROID_HOME=${ANDROID_HOME}"; \
+    echo "export ANDROID_NDK_HOME=${ANDROID_NDK_HOME}"; \
+    echo "export JAVA_HOME=${JAVA_HOME}"; \
+    echo "export ANDROID_CMDLINE_TOOLS_PATH=${ANDROID_CMDLINE_TOOLS_PATH}"; \
+    echo "export PATH=${PATH}"; \
+  } > /usr/bin/unity-editor.d/android-2019+.sh
+
+#=======================================================================================
+# [2019.x/2020.x-android] Accept Android SDK licenses via old sdkmanager
+#=======================================================================================
+RUN echo "$version-$module" | grep -q -vP '^(20(19|20)).*android' \
+  && exit 0 \
+  || : \
+  && . ~/.bashrc \
+  && yes | "${ANDROID_HOME}/tools/bin/sdkmanager" --licenses
+
+#=======================================================================================
+# [2021.x/2022.x-android] Accept Android SDK licenses via new cmdline-tools sdkmanager
+#=======================================================================================
+RUN echo "$version-$module" | grep -q -vP '^(202[1-9]|20[3-9]).*android' \
+  && exit 0 \
+  || : \
+  && . ~/.bashrc \
+  && yes | "${ANDROID_CMDLINE_TOOLS_PATH}/bin/sdkmanager" --licenses
+
+#=======================================================================================
+# [2022.x-android] Fix for symlink issue on Android 2022.2+
+#=======================================================================================
+RUN echo "$version-$module" | grep -q -vP '^(2022.[2-9]|202[3-9]|20[3-9]).*android' \
+  && exit 0 \
+  || : \
+  && . ~/.bashrc \
+  && cd "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin" \
+  # Symlink any file less than 64 bytes to the file name within the file. We assume there are no real files that small
+  && for f in $(find . -type f -size -64c); do target=$(cat $f) && echo "Making symlink $f -> $target" && rm $f && ln -s $target $f ; done
```

Comment:
Why are we using file size here for the loop? Is it to identify files that are symlinks?
If so, we can do so by using the `-L` test  to determine whether it is a symbolic link. 

Example:

```bash
RUN echo "$version-$module" | grep -q -vP '^(2022.[2-9]|202[3-9]|20[3-9]).*android' \
  && exit 0 \
  || : \
  && . ~/.bashrc \
  && cd "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin" \
  && for f in *; do \
       if [ -L "$f" ]; then \
         target=$(readlink "$f") && echo "Making symlink $f -> $target" && rm "$f" && ln -s "$target" "$f"; \
       fi \
     done
```

_I haven't tried this_

Reply:
So the files aren't actually symbolic links, they're a mix of ascii and binary files but inside they have the file name of the executable they should point to. We suspect hub isn't using the proper flags when unzipping so the symbolic links get lost hence the need for us to make the links ourselves.

Reply:
Possibly more manageable to separate that part into a different PR? Not sure though. And not preference from my side.

Reply:
This was the main point of the PR to fix the android build issues of 2022.2+ so I'm inclined to keep this as a single PR.

Reply:
Yea, makes sense!